src/vtcomposite.cpp

```diff
@@ -692,104 +716,183 @@ struct LocalizeWorker : Napi::AsyncWorker
                 vtzero::layer_builder lbuilder{tbuilder, layer.name(), layer.version(), layer.extent()};
                 while (auto feature = layer.next_feature())
                 {
-                    std::vector<std::string> worldviews_to_create;
+                    // will be skipping features with incompatible worldview
+                    bool skip_feature = false;
+
+                    // will be creating one clone of the feature for each worldview if worldview property exists
                     bool has_worldview_key = false;
-                    bool name_was_set = false;
-                    vtzero::property_value name_value;
+                    std::vector<std::string> worldviews_to_create;
+
+                    // will be searching for the class with lowest index in class_key_precedence
+                    auto class_key_idx = static_cast<std::uint32_t>(class_key_precedence.size());
                     vtzero::property_value class_value;
-                    // accumulate final properties (except _mbx_worldview translation to worldview) here
-                    std::vector<std::pair<std::string, vtzero::property_value>> properties;
+
+                    auto language_key_idx = static_cast<std::uint32_t>(language_key_precedence.size());
+                    vtzero::property_value language_value;
+                    vtzero::property_value original_language_value;
+
+                    // collect final properties
+                    std::vector<std::pair<std::string, vtzero::property_value>> final_properties;
                     while (auto property = feature.next_property())
                     {
+                        // if already know we'll be skipping this feature, don't need to comb through its properties
+                        if (skip_feature)
```

Comment:
could this added to the while? `while (property = next_property && !skip_feature)`

Reply:
Unfortunately no, that turns `property` into bool type.

Reply:
To be clear I meant something like
   `while (!skip_feature && auto feature = layer.next_feature())`
rather than have the skip_feature check on L739.
It's not a very important detail but would be a little more readable IMO

Reply:
The while loop `while (auto feature = layer.next_feature())` is for looping through the features one by one; we don't know whether we can skip a feature or not before we look at its properties, so `skip_feature` is always false when we evaluate this while condition. The `skip_feature` flag changes to true when we encounter a property that suggests we can discard this feature, and the flag resets to false when we are done with one feature and moves on to the next feature. I'll add a comment to clarify this.

`while (!skip_feature && auto property = feature.next_property())` results in this error:
```
../src/vtcomposite.cpp:736:45: error: expected expression
                    while (!skip_feature && auto property = feature.next_property())
                                            ^
../src/vtcomposite.cpp:744:52: error: use of undeclared identifier 'property'
                        std::string property_key = property.key().to_string();
                                                   ^
``` 
and `while (auto property = feature.next_property() && !skip_feature)`  results in this error:
```
../src/vtcomposite.cpp:744:60: error: member reference base type 'bool' is not a structure or union
                        std::string property_key = property.key().to_string();
                                                   ~~~~~~~~^~~~
```

Reply:
Added more details to the comments re: `skip_feature` in https://github.com/mapbox/vtcomposite/pull/129/commits/759e7099809cd65f787ea92c3b7af739da7d2b7c.